emission/analysis/classification/inference/mode.py

```diff
@@ -191,47 +271,53 @@ def getGroundTruthMode(self, section):
 # 20. both start and end close to train station
 # 21. both start and end close to airport
   def updateFeatureMatrixRowWithSection(self, featureMatrix, i, section):
-    featureMatrix[i, 0] = section['distance']
-    featureMatrix[i, 1] = (section['section_end_datetime'] - section['section_start_datetime']).total_seconds()
+    if 'distance' in section.keys():
+      featureMatrix[i,0] = section['distance']
+    else:
+      featureMatrix[i, 0] = easf.calDistance(section['start_loc']['coordinates'], section['end_loc']['coordinates'])  #####would need to calculate this
+
+    featureMatrix[i, 1] = (section['end_ts'] - section['start_ts'])
 
     # Deal with unknown modes like "airplane"
     try:
-      featureMatrix[i, 2] = section['mode']
+      featureMatrix[i, 2] = section['sensed_mode']
     except ValueError:
       featureMatrix[i, 2] = 0
 
-    featureMatrix[i, 3] = section['section_id']
-    featureMatrix[i, 4] = easf.calAvgSpeed(section)
-    speeds = easf.calSpeeds(section)
-    if speeds != None and len(speeds) > 0:
+    featureMatrix[i, 3] = section['section_id']   ####
+    #featureMatrix[i, 4] = easf.calAvgSpeed(section)
```

Comment:
It isn't. I commented it out to see how it was working and simply forgot to put it back in. Since the entry should be zero anyway, it wasn't obvious I forgot that bit.